libphx/src/shader.rs

```diff
@@ -228,16 +246,21 @@ pub unsafe extern "C" fn Shader_Load(
     fName: *const libc::c_char,
 ) -> Box<Shader> {
     let mut this = Box::new(Shader::default());
+
```

Comment:
Same here, this could be `Shader::load` that returns a `Shader`, then we `Box` it in `Shader_Load` for C.